probe/docker/controls.go

```diff
@@ -15,69 +17,76 @@ const (
-	log.Printf("Stopping container %s", req.NodeID)
-
-	_, containerID, ok := report.ParseContainerNodeID(req.NodeID)
-	if !ok {
-		return xfer.ResponseErrorf("Invalid ID: %s", req.NodeID)
-	}
-
+func (r *registry) stopContainer(containerID string, _ xfer.Request) xfer.Response {
+	log.Printf("Stopping container %s", containerID)
 	return xfer.ResponseError(r.client.StopContainer(containerID, waitTime))
 }
 
-func (r *registry) startContainer(req xfer.Request) xfer.Response {
-	log.Printf("Starting container %s", req.NodeID)
-
-	_, containerID, ok := report.ParseContainerNodeID(req.NodeID)
-	if !ok {
-		return xfer.ResponseErrorf("Invalid ID: %s", req.NodeID)
-	}
-
+func (r *registry) startContainer(containerID string, _ xfer.Request) xfer.Response {
+	log.Printf("Starting container %s", containerID)
 	return xfer.ResponseError(r.client.StartContainer(containerID, nil))
 }
 
-func (r *registry) restartContainer(req xfer.Request) xfer.Response {
-	log.Printf("Restarting container %s", req.NodeID)
-
-	_, containerID, ok := report.ParseContainerNodeID(req.NodeID)
-	if !ok {
-		return xfer.ResponseErrorf("Invalid ID: %s", req.NodeID)
-	}
-
+func (r *registry) restartContainer(containerID string, _ xfer.Request) xfer.Response {
+	log.Printf("Restarting container %s", containerID)
 	return xfer.ResponseError(r.client.RestartContainer(containerID, waitTime))
 }
 
-func (r *registry) pauseContainer(req xfer.Request) xfer.Response {
-	log.Printf("Pausing container %s", req.NodeID)
-
-	_, containerID, ok := report.ParseContainerNodeID(req.NodeID)
-	if !ok {
-		return xfer.ResponseErrorf("Invalid ID: %s", req.NodeID)
-	}
-
+func (r *registry) pauseContainer(containerID string, _ xfer.Request) xfer.Response {
+	log.Printf("Pausing container %s", containerID)
 	return xfer.ResponseError(r.client.PauseContainer(containerID))
 }
 
-func (r *registry) unpauseContainer(req xfer.Request) xfer.Response {
-	log.Printf("Unpausing container %s", req.NodeID)
+func (r *registry) unpauseContainer(containerID string, _ xfer.Request) xfer.Response {
+	log.Printf("Unpausing container %s", containerID)
+	return xfer.ResponseError(r.client.UnpauseContainer(containerID))
+}
 
-	_, containerID, ok := report.ParseContainerNodeID(req.NodeID)
-	if !ok {
-		return xfer.ResponseErrorf("Invalid ID: %s", req.NodeID)
+func (r *registry) attachContainer(containerID string, req xfer.Request) xfer.Response {
+	pipe, err := controls.NewPipe(req.AppID, func() {})
+	if err != nil {
+		xfer.ResponseError(err)
 	}
+	go func() {
+		if err := r.client.AttachToContainer(docker_client.AttachToContainerOptions{
+			Container:    containerID,
+			RawTerminal:  true,
+			Stream:       true,
+			Stdin:        true,
+			Stdout:       true,
+			Stderr:       true,
+			InputStream:  pipe.Starboard(),
+			OutputStream: pipe.Starboard(),
+			ErrorStream:  pipe.Starboard(),
+		}); err != nil {
+			log.Printf("Error from docker attach: %v", err)
+		}
```

Comment:
Should probably return this error to the requester somehow, no? Could we do this synchonously and return the error up front?